### ISSUE TITLE
Auto-indent is weird...

### ISSUE THREAD

**Issue by TangerineGameDev** [luckasRanarison/tree-sitter-hypr#2]
https://github.com/luckasRanarison/tree-sitter-hypr/assets/81909104/eb4d2e7b-a6d4-464c-a3de-f323d295a1e6

**Comment by luckasRanarison** [comment]
Is this indentation using `>` ? I'm not sure if it's really related, by indentation feature, I meant the auto indent when jumping to a new line. But indentation using `>` works for me, can you provide the actual file?

**Comment by TangerineGameDev** [comment]
Well indentation using `>` is a neovim feature afaik, which just takes your set tab size (mine is 4 spaces for each tab) and decreases or increases the amount of tabs placed. It doesn't have to do with **auto**-indent or tree-sitter, etc.

Auto-Indent is whenever you press `==` on a line in normal mode, or `=` in visual line mode.
It's supposed to detect the files structure (brackets, curly braces, etc) and **auto**-indent a line. (I'm not sure though, if it's supposed to be implemented in the code here)

In my case, my issue lies in how I have a binding that I always auto-indents the whole file and a habit of mine is doing it every time I save... (what the binding does: `ggVG=`

If this could be fixed, I would be eternally grateful =)

**Comment by luckasRanarison** [comment]
I didn't know about that auto-indent, maybe this issue was caused by some ERROR nodes in the trees because of the linebreaks. I fixed that issue in 0050af1, could you try if the auto-indentation issue was also fixed? (Also don't forget to update plugin)

**Comment by TangerineGameDev** [comment]
I did `:TSInstall hypr` and `:PackerSync`, still causes the same issue...

Could this be caused from nested brackets?

This is how these lines look, whenever I auto-indent, notice the nested brackets (touchpad):
```
input {
    kb_layout = de,us
        kb_variant =
        kb_model =
        kb_options = grp:ctrl_space_toggle
        kb_rules =

        repeat_delay = 300
        repeat_rate = 30

        follow_mouse = 1

        touchpad {
            natural_scroll = yes
                scroll_factor = 0.38
        }

    sensitivity = 0.78 # -1.0 - 1.0, 0 means no modification.
        accel_profile = flat
}

```

**Comment by TangerineGameDev** [comment]
Also just now I noticed, that auto-indenting the whole file causes bindings at the bottom to skyrocket XD

Whole config:
```
####################
# Main config file #
####################

# Import enviroment variables
source = ~/.config/hypr/env/backend.conf
source = ~/.config/hypr/env/nvidia.conf
source = ~/.config/hypr/env/theme.conf
source = ~/.config/hypr/env/xdg.conf
source = ~/.config/hypr/env/qt.conf

# See https://wiki.hyprland.org/Configuring/Monitors/
monitor=Main-Display,highrr,auto,auto,bitdepth,10

# See https://wiki.hyprland.org/Configuring/Keywords/ for more

# Execute your favorite apps at launch
# exec-once = waybar & hyprpaper & firefox
exec-once = ulauncher --no-window-shadow --hide-window
#exec-once = dunst
exec-once = swaync
exec-once = /usr/lib/polkit-kde-authentication-agent-1
exec-once = udiskie &
exec-once = wl-clip-persist --clipboard both
exec-once = dbus-update-activation-environment --systemd WAYLAND_DISPLAY XDG_CURRENT_DESKTOP
exec-once = hyprctl setcursor Bibata-Modern-Classic 24
exec-once = swayidle -w timeout 300 'swaylock --fade-in 1 --grace 10 -S --clock --effect-blur 10x5'
exec-once = wljoywake

exec-once = sleep 2 ; swww init & swww img ~/.local/wallpapers/Kafka Boom.jpg

# Open Discord minimized on startup
exec-once = discord --start-minimized

# For all categories, see https://wiki.hyprland.org/Configuring/Variables/
input {
    kb_layout = de,us
        kb_variant =
        kb_model =
        kb_options = grp:ctrl_space_toggle
        kb_rules =

        repeat_delay = 300
        repeat_rate = 30

        follow_mouse = 1

        touchpad {
            natural_scroll = yes
                scroll_factor = 0.38
        }

    sensitivity = 0.78 # -1.0 - 1.0, 0 means no modification.
        accel_profile = flat
}

general {
# See https://wiki.hyprland.org/Configuring/Variables/ for more

    gaps_in = 2
        gaps_out = 7
        border_size = 3
        col.active_border = rgba(33ccffee) rgba(ff54a6ee) 55deg
        col.inactive_border = rgba(acacac80)

        hover_icon_on_border = off

        layout = dwindle
}

decoration {
# See https://wiki.hyprland.org/Configuring/Variables/ for more

    inactive_opacity = 1.95

        rounding = 5

        drop_shadow = no
        shadow_range = 30
        shadow_render_power = 4
        col.shadow = rgba(1a1a1aee)

        blur {
            enabled = yes
                size = 3
                passes = 2
                xray = on
                brightness = 1
                new_optimizations = on
        }
}

animations {
    enabled = yes

# Some default animations, see https://wiki.hyprland.org/Configuring/Animations/ for more


        bezier = myBezier, 0.05, 0.9, 0.1, 1.05
        bezier = easeInOut, 0.5, 1.0, 0.89, 1.0
        bezier = linear, 0.5, 0.5, 0.5, 0.5
        bezier = workspaceScroll, 0.5, 1, 0.89, 1

        animation = windows, 1, 7, myBezier, slide
        animation = windowsMove, 1, 3, myBezier
        animation = windowsOut, 1, 7, default, popin 80%
        animation = border, 1, 10, default
        animation = borderangle, 1, 8, default
        animation = fade, 1, 7, default
        animation = workspaces, 1, 6, default
        animation = borderangle, 1, 30, easeInOut, once
#animation = workspaces, 1, 5, workspaceScroll
}

dwindle {
# See https://wiki.hyprland.org/Configuring/Dwindle-Layout/ for more
    pseudotile = yes # master switch for pseudotiling. Enabling is bound to mainMod + P in the keybinds section below
        preserve_split = yes # you probably want this+
        smart_split = yes
}

master {
# See https://wiki.hyprland.org/Configuring/Master-Layout/ for more
    new_is_master = true
        orientation = center
#always_center_master = true
}

gestures {
# See https://wiki.hyprland.org/Configuring/Variables/ for more
    workspace_swipe = on
        workspace_swipe_distance = 200
        workspace_swipe_forever = true
        workspace_swipe_cancel_ratio = 0.25
}

misc {
    disable_hyprland_logo = off
        mouse_move_enables_dpms = on
        focus_on_activate = true
}

binds {
    scroll_event_delay = 0
}

# Example per-device config
# See https://wiki.hyprland.org/Configuring/Keywords/#executing for more
# device:epic-mouse-v1 {
#     sensitivity = -0.5
# }

device:elan050a:01-04f3:3158-touchpad {
           sensitivity = 0.5 # -1.0 - 1.0, 0 means no modification.
               natural_scroll = on
       }

# Example windowrule v1
# windowrule = float, ^(kitty)$
# Example windowrule v2
# windowrulev2 = float,class:^(kitty)$,title:^(kitty)$
# See https://wiki.hyprland.org/Configuring/Window-Rules/ for more
windowrulev2 = rounding 20,title:^(.*Ulauncher.*)$
windowrulev2 = dimaround,title:^(.*Ulauncher.*)$

windowrulev2 = float,title:^(.*Authentication Required — PolicyKit1 KDE Agent.*)$
windowrulev2 = float,title:^(.*File Operation Progress.*)$
windowrulev2 = float,title:^(.*Picture in Picture.*)$
windowrulev2 = float,class:^(.*zenity.*)$
                               windowrulev2 = float,class:^(.*kmix.*)$

                                                              windowrulev2 = workspace special,class:^(discord)$

                                                                                                       windowrulev2 = float,class:^(.*pavucontrol.*)$
                                                                                                                                      windowrulev2 = size 562 400,class:^(.*pavucontrol.*)$
                                                                                                                                                                            windowrulev2 = move 100%-592 100%-430,class:^(.*pavucontrol.*)$

# See https://wiki.hyprland.org/Configuring/Keywords/ for more
                                                                                                                                                                                                                            $mainMod = SUPER
                                                                                                                                                                                                                            $mainModShift = SUPERSHIFT

# Example binds, see https://wiki.hyprland.org/Configuring/Binds/ for more
                                                                                                                                                                                                                            bind = $mainMod, T, exec, alacritty
                                                                                                                                                                                                                            bind = $mainMod, Q, killactive, # exec, . ~/.config/hypr/minimizable-kill.sh 
                                                                                                                                                                                                                            bind = $mainMod, E, exec, dolphin
                                                                                                                                                                                                                            bind = $mainMod, V, togglefloating,
#bind = $mainMod, Space, exec, ulauncher-toggle # Open App Launcher
#bind = $mainMod, Space, exec, pkill rofi || rofi -show drun -matching fuzzy -sorting-method fzf -sort # Open App Launcher
                                                                                                                                                                                                                            bind = $mainMod, Space, exec, sirula
                                                                                                                                                                                                                            bind = $mainMod, P, pseudo, # dwindle
                                                                                                                                                                                                                            bind = $mainMod, J, togglesplit, # dwindle
                                                                                                                                                                                                                            bind = $mainMod, F, fullscreen, 0
                                                                                                                                                                                                                            bind = $mainModShift, F, fullscreen, 1
                                                                                                                                                                                                                            bind = $mainMod, escape, exec, wlogout -p layer-shell

# Take a screenshot and save it to the clipboard and $HOME/Pictures
                                                                                                                                                                                                                            bind = , code:107, exec, mkdir -p ~/Pictures | slurp | grim -g - - | wl-copy && wl-paste > ~/Pictures/$(date +'%d.%m.%Y %H:%M:%S.png')

# Move focus with mainMod + arrow keys
                                                                                                                                                                                                                            bind = $mainMod, left, movefocus, l
                                                                                                                                                                                                                            bind = $mainMod, right, movefocus, r
                                                                                                                                                                                                                            bind = $mainMod, up, movefocus, u
                                                                                                                                                                                                                            bind = $mainMod, down, movefocus, d

# Switch workspaces with mainMod + [0-9]
                                                                                                                                                                                                                            bind = $mainMod, 1, workspace, 1
                                                                                                                                                                                                                            bind = $mainMod, 2, workspace, 2
                                                                                                                                                                                                                            bind = $mainMod, 3, workspace, 3
                                                                                                                                                                                                                            bind = $mainMod, 4, workspace, 4
                                                                                                                                                                                                                            bind = $mainMod, 5, workspace, 5
                                                                                                                                                                                                                            bind = $mainMod, 6, workspace, 6
                                                                                                                                                                                                                            bind = $mainMod, 7, workspace, 7
                                                                                                                                                                                                                            bind = $mainMod, 8, workspace, 8
                                                                                                                                                                                                                            bind = $mainMod, 9, workspace, 9
                                                                                                                                                                                                                            bind = $mainMod, 0, workspace, 10

# Move active window to a workspace with mainMod + SHIFT + [0-9]

                                                                                                                                                                                                                            bind = $mainModShift, 1, movetoworkspace, 1
                                                                                                                                                                                                                            bind = $mainModShift, 2, movetoworkspace, 2
                                                                                                                                                                                                                            bind = $mainModShift, 3, movetoworkspace, 3
                                                                                                                                                                                                                            bind = $mainModShift, 4, movetoworkspace, 4
                                                                                                                                                                                                                            bind = $mainModShift, 5, movetoworkspace, 5
                                                                                                                                                                                                                            bind = $mainModShift, 6, movetoworkspace, 6
                                                                                                                                                                                                                            bind = $mainModShift, 7, movetoworkspace, 7
                                                                                                                                                                                                                            bind = $mainModShift, 8, movetoworkspace, 8
                                                                                                                                                                                                                            bind = $mainModShift, 9, movetoworkspace, 9
                                                                                                                                                                                                                            bind = $mainModShift, 0, movetoworkspace, 10

# Scroll through existing workspaces with mainMod + scroll
                                                                                                                                                                                                                            bind = $mainMod, mouse_down, workspace, e+1
                                                                                                                                                                                                                            bind = $mainMod, mouse_up, workspace, e-1

# Move/resize windows with mainMod + LMB/RMB and dragging
                                                                                                                                                                                                                            bindm = $mainMod, mouse:272, movewindow
                                                                                                                                                                                                                            bindm = $mainMod, mouse:273, resizewindow

# Move windows with mainMod + Shift + Arrow
                                                                                                                                                                                                                            bind = $mainModShift, left, movewindow, l
                                                                                                                                                                                                                            bind = $mainModShift, right, movewindow, r
                                                                                                                                                                                                                            bind = $mainModShift, up, movewindow, u
                                                                                                                                                                                                                            bind = $mainModShift, down, movewindow, d


# Change volume
#bindle =, XF86AudioRaiseVolume, exec, amixer -q set Master 5%+
#bindle =, XF86AudioLowerVolume, exec, amixer -q set Master 5%-
#bindle =, XF86AudioMute, exec, amixer -q -D pulse set Master 1+ toggle
                                                                                                                                                                                                                            bindle = Alt, F3, exec, swayosd --output-volume lower
                                                                                                                                                                                                                            bindle = Alt, F4, exec, swayosd --output-volume raise
                                                                                                                                                                                                                            bindle = Alt, F5, exec, swayosd --output-volume mute-toggle

# Change screen brightness
                                                                                                                                                                                                                            bindle = Alt, F1, exec, brillo -u 150000 -U 5
                                                                                                                                                                                                                            bindle = Alt, F2, exec, brillo -u 150000 -A 5

# Open Notification Center
                                                                                                                                                                                                                            bind = $mainMod, N, exec, swaync-client -t -sw

# Special workspace support for hiding windows (Spotify, etc...)
                                                                                                                                                                                                                            bind = $mainModShift, X, movetoworkspace, special
                                                                                                                                                                                                                            bind = $mainModShift, Y, togglespecialworkspace,

                                                                                                                                                                                                                            bind = Alt+Shift, V, exec, pavucontrol


                                                                                                                                                                                                                            bindr = , XF86Presentation, exec, . ~/.config/hypr/toggle-fans-max.sh

```

**Comment by luckasRanarison** [comment]
Really weird, it's working for me. I don't know if it's relevant but I'm using neovim nightly.

https://github.com/luckasRanarison/tree-sitter-hypr/assets/101930730/7995dfca-73b6-48d2-9b2c-0879c3369f68

**Comment by TangerineGameDev** [comment]
That is seriously weird...
Even running Neovim without any enviroment (using sudo without the -E flag disables all plugins, settings, etc.) still yields the same results...

Maybe I have installed something incorrectly, then again I followed the README to the t...

Maybe you have some kind of plugin that allows auto bracket indenting?

I'll public my .dotfiles so that you can take a look and compare them...
(They are supposed to be public, but are a work in progress :P)

**Comment by TangerineGameDev** [comment]
Here you go: https://github.com/TangerineGameDev/.dotfiles

**Comment by TangerineGameDev** [comment]
I found it...

**Comment by luckasRanarison** [comment]
> I found it...

???

**Comment by TangerineGameDev** [comment]
I'm sorry I'm quite new and as I said I'm setting up my desktop enviroment...
I followed ThePrimeagens Vid for Neovim as a starter point, and it seems like he didn't mention how to enable indenting in treesitter.

I thought it would be on by default? (Why wouldn't it be...)

Anyways...
I added this line to `require'nvim-treesitter.configs'.setup` and now it works.
```lua
  indent = {
    enable = true,
    disable = {"python"}
  },
```

I'm sorry for my inexperience...
